manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -94,16 +101,16 @@ func (tr *TaskReaper) Run() {
 }
 
 func (tr *TaskReaper) tick() {
-	if len(tr.dirty) == 0 {
+	if len(tr.dirty) == 0 && len(tr.orphaned) == 0 {
 		return
 	}
 
 	defer func() {
 		tr.dirty = make(map[instanceTuple]struct{})
+		tr.orphaned = nil
 	}()
 
-	var deleteTasks []string
-
+	deleteTasks := tr.orphaned
```

Comment:
>  If it is fine to delete tasks from anywhere then we can do away with ORPHANED state.

There are also cases where we may want to keep orphaned tasks around for record keeping.

I think having tasks be deleted in one place is a great property.